src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -434,14 +434,24 @@ private void InitializeHttpFunctions(IEnumerable<FunctionDescriptor> functions,
             }
         }
 
-        public override void Shutdown()
+        public override void Shutdown(bool hard = false)
         {
             string message = "Environment shutdown has been triggered. Stopping host and signaling shutdown.";
             Instance?.TraceWriter.Info(message);
             Instance?.Logger?.LogInformation(message);
 
             Stop();
-            HostingEnvironment.InitiateShutdown();
+
+            if (hard)
+            {
+                // "hard" shutdown recycles the process
+                Process.GetCurrentProcess().Kill();
+            }
+            else
+            {
+                // "soft" shutdown recycles the AppDomain
+                HostingEnvironment.InitiateShutdown();
```

Comment:
This soft recycle remains for other shutdown paths, e.g. after nuget package updates, etc.